lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -523,7 +561,9 @@ LowererMDArch::LowerCallArgs(IR::Instr *callInstr, ushort callFlags, Js::ArgSlot
         callInstr->InsertBefore(argInstr);
         argCount++;
     }
-
+#ifndef _WIN32
+    this->xplatCallArgs.StopRecording();
```

Comment:
If you follow the usages of `GetArgSlotOpnd` you will see that there are 2 places setting argSym so the register can be anything but `TyMachReg`. One of them we record. Other, we don't have to since `shouldHomeParams = argCount > 0` and `!shouldHomeParams`... So we record them all right.